naturewatch_camera_server/CameraController.py

```diff
@@ -110,129 +110,147 @@ def stop(self):
+            self.picamera_video_stream.clear()
+            self.camera.start_recording(self.picamera_video_stream, format='h264', bitrate=self.video_bitrate)
+            self.logger.debug('CameraController: recording started')
 
-    def stop_circular_stream(self):
+    def stop_video_stream(self):
         if picamera_exists:
             self.camera.stop_recording()
+            self.logger.debug('CameraController: recording stopped')
 
     def wait_recording(self, delay):
         if picamera_exists:
             return self.camera.wait_recording(delay)
         else:
             return None
 
-    def clear_buffer(self):
-        if picamera_exists:
-            self.circularStream.clear()
-
+    # TODO: Not used?
     def get_thumb_image(self):
-        self.logger.info("Requested thumb image.")
+        self.logger.debug("CameraController: lores image requested.")
         if picamera_exists:
             return self.get_image_binary()
         else:
             return None
 
-    # Get splitter image
-    def get_splitter_image(self):
-        self.logger.info("Requested splitter image.")
-        if self.use_splitter_port:
-            if picamera_exists:
-                stream = io.BytesIO()
-                self.camera.capture(stream, format='jpeg', use_video_port=True)
-                stream.seek(0)
-                data = np.fromstring(stream.getvalue(), dtype=np.uint8)
-                # "Decode" the image from the array, preserving colour
-                s = cv2.imdecode(data, 1)
-
-                if s is not None:
-                    return s.copy()
-                else:
-                    return None
-
+    # Get high res image
+    def get_hires_image(self):
+        self.logger.debug("CameraController: hires image requested.")
+        if picamera_exists:
+            # TODO: understand the decode. Is another more intuitive way possible?
+            self.picamera_photo_stream = io.BytesIO()
+            self.camera.capture(self.picamera_photo_stream, format='jpeg', use_video_port=self.use_video_port)
+            self.picamera_photo_stream.seek(0)
+            # "Decode" the image from the stream, preserving colour
+            s = cv2.imdecode(np.fromstring(self.picamera_photo_stream.getvalue(), dtype=np.uint8), 1)
+
+            if s is not None:
+                return s.copy()
             else:
-                if self.splitter_image is not None:
-                    return self.splitter_image.copy()
-                else:
-                    return None
+                return None
+
         else:
-            self.logger.warning("Splitter image was not opened in constructor.")
-            return None
+            if self.hires_image is not None:
+                return self.hires_image.copy()
+            else:
+                return None
 
     # Initialise picamera. If already started, close and reinitialise.
     # TODO - reset with manual exposure, if it was set before.
```

Comment:
I am not sure what this TODO means

Reply:
Here's what we were thinking: if the user has chosen manual exposure and the camera crashes and reinitialises, it should be reset with manual exposure settings the user had chosen before.